Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedMapModel.swift

```diff
@@ -59,19 +104,96 @@ class PreplannedMapModel: ObservableObject, Identifiable {
             status = .packageFailure
         case .complete:
             status = .packaged
-        @unknown default:
-            fatalError("Unknown packaging status")
         }
     }
     
-    /// A Boolean value indicating if download can be called.
-    var canDownload: Bool {
-        switch status {
-        case .notLoaded, .loading, .loadFailure, .packaging, .packageFailure,
-                .downloading, .downloaded:
-            false
-        case .packaged, .downloadFailure:
-            true
+    /// Updates the status based on the download result of the mobile map package.
+    func updateDownloadStatus(for downloadResult: Result<DownloadPreplannedOfflineMapResult, any Error>?) {
+        switch downloadResult {
+        case .success:
+            status = .downloaded
+        case .failure(let error):
+            status = .downloadFailure(error)
+        case .none:
+            return
+        }
+    }
+    
+    /// Looks in the  mobile map package if downloaded locally.
+    private func lookupMobileMapPackage() -> MobileMapPackage? {
+        let fileURL = FileManager.default.preplannedDirectory(
+            forPortalItemID: portalItemID,
+            preplannedMapAreaID: preplannedMapAreaID
+        )
+        guard FileManager.default.fileExists(atPath: fileURL.relativePath) else { return nil }
+        return MobileMapPackage.init(fileURL: fileURL)
+    }
+    
+    /// Posts a local notification that the job completed with success or failure.
+    /// - Precondition: `job.status == .succeeded || job.status == .failed`
+    private static func notifyJobCompleted(job: DownloadPreplannedOfflineMapJob) async throws {
+        precondition(job.status == .succeeded || job.status == .failed)
+        guard
+            let preplannedMapArea = job.parameters.preplannedMapArea,
+            let id = preplannedMapArea.id
+        else { return }
+        
+        let content = UNMutableNotificationContent()
+        content.sound = UNNotificationSound.default
+        
+        let jobStatus = job.status == .succeeded ? "Succeeded" : "Failed"
+        
+        content.title = "Download \(jobStatus)"
+        content.body = "The job for \(preplannedMapArea.title) has \(jobStatus.lowercased())."
```

Comment:
Do you want to change them to localized [String](https://developer.apple.com/documentation/swift/string/init(localized:table:bundle:locale:comment:)) so we can keep track of them for future localization? 

Did we use the formatting argument placeholder like `%@` before in localization process?

Reply:
I will make this change in a separate PR to support localization.